File: gpt-4/template-context-full/kernels_construct_firstprivate_clause.c
#include "acc_testsuite.h"
#include <stdlib.h>

#ifndef T1
/*T1:kernels construct firstprivate clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);
    int *a = (int*)malloc(10*sizeof(int));
    int *b = (int*)malloc(10*sizeof(int));
    
    #pragma acc data create(a[0:10]) copy(b[0:10])
    {
        #pragma acc update device(a[0:10])
        
        for(int i=0; i<10; i++)
        {
            a[i]=rand()%100;
        }

        int preview_a = a[5];

        #pragma acc kernels firstprivate(preview_a)
        {
            for(int i=0; i<10; i++)
            {
                b[i] = a[i] + preview_a;
            }
        }

        #pragma acc update host(b[0:10])

        for(int i=0; i<10; i++)
        {
            if(b[i] != a[i] + preview_a){
                err = 1;
            }
        }
    }

    free(a);
    free(b);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/template-context-full/kernels_construct_firstprivate_clause.f90
#ifndef T1
!T1:kernels construct firstprivate clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(KIND=8),DIMENSION(10)::a, b
        REAL(KIND=8):: c


        !$acc data copy(a(1:10)) 
          c = 1
          !$acc kernels firstprivate(c)
          !$acc loop 
          DO test = 1, 600
		
            !$acc loop independent
            DO i = 1, 10
            c = 1
              a(i) = a(i) + c
            END DO
            
          END DO    
          !$acc end kernels

        !$acc end data

        DO i = 1, 10
          x = x + a(i)
        END DO
        IF x .eq. 0 THEN 
          errors = errors + 1
        ENDIF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM